subprojects/plugins/src/test/groovy/org/gradle/api/tasks/testing/TestTaskSpec.groovy

```diff
@@ -29,12 +29,14 @@ class TestTaskSpec extends AbstractProjectBuilderSpec {
     def testFramework = Mock(TestFramework)
     def suiteDescriptor = Mock(TestDescriptorInternal)
     def testDescriptor = Mock(TestDescriptorInternal)
+    def testBuildOperationAdapter = Mock(TestListenerBuildOperationAdapter)
 
     private WorkerLeaseRegistry.WorkerLeaseCompletion completion
     private Test task
 
     def setup() {
         task = TestUtil.create(temporaryFolder).task(Test, [testExecuter: testExecuter, testFramework: testFramework])
+        task.testListenerBuildOperationAdapter = testBuildOperationAdapter
```

Comment:
Why is this necessary?

Reply:
not mocking would result in a `java.lang.IllegalStateException: No operation is currently running.`
thrown by the `DefaultBuildOperationExecutor` when calling `getBuildOperationExecutor().getCurrentOperation()`.

I guess the alternative would be to change `ProjectBuilderImpl` which I thought is too invasive.